tools/crates/build_crate.gni

```diff
@@ -36,6 +36,8 @@ template("build_crate") {
     #   --frozen (--locked + --offline)
     # when running `cargo build`.
     args = [
+      "--temp_dir_path",
+      rebase_path(root_out_dir),
```

Comment:
I would either use `target_out_dir` here or give it some path within `root_out_dir`. This also needs to be in the outputs list to ensure that there are no conflicts with outputs from other targets. 

I also noticed below that there are unlisted inputs here like `Cargo.toml` and `.cargo/config.toml` and I'm a bit concerned that other artifacts are left in `target_dir` that are not listed as outputs and may therefore also conflict. I don't think it's a good idea to have any of these outputs in the root out dir for exactly this reason.

Reply:
also ideally it should generate a depfile with `--emit=dep-info` 
https://gn.googlesource.com/gn/+/main/docs/reference.md#var_depfile